src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -88,15 +101,13 @@ export const RequestPage = () => {
   const toSetupPage = useCallback(() => {
     const referrer = extractReferrerFromSearchParameters(searchParams)
     const isNewOnboardingFlowEnabled = flags[FeatureFlagsKeys.NEW_ONBOARDING_FLOW]
-    const isIos = userAgentData?.os.name === 'iOS'
-    const isSafari = !isLoadingUserAgentData && userAgentData?.browser.name === 'Safari'
-    const isAvatarSetupFlowAllowed = isNewOnboardingFlowEnabled && !isIos && !isSafari
+    const isAvatarSetupFlowAllowed = isNewOnboardingFlowEnabled && hasWebGPU && !isCheckingWebGPU
```

Comment:
What if instead of using the hook, we use the `checkWebGpuSupport` function here?
We can make the `toSetupPage` an async function, as it's called inside of another async function.